cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs

```diff
@@ -666,96 +667,365 @@ fn parse_url(
+            },
+            TestCase {
+                input: "quay.io/".to_string(),
+                expected_registry: Registry {
                     scheme: "".to_string(),
                     insecure: false,
                     host: "quay.io".to_string(),
                     port: None,
-                    namespace: "".to_string(),
+                    namespace: "/".to_string(),
+                },
+                expected_registry_host_port_string: "quay.io".to_string(),
+                expected_registry_host_port_namespaced_string: "quay.io/".to_string(),
+            },
+            TestCase {
+                input: "quay.io:8080".to_string(),
+                expected_registry: Registry {
+                    scheme: "".to_string(),
+                    insecure: false,
+                    host: "quay.io".to_string(),
+                    port: Some(8080),
+                    namespace: "/".to_string(),
                 },
-            ),
-            (
-                "quay.io/ns1",
-                Registry {
+                expected_registry_host_port_string: "quay.io:8080".to_string(),
+                expected_registry_host_port_namespaced_string: "quay.io:8080/".to_string(),
+            },
+            TestCase {
+                input: "quay.io/ns1".to_string(),
+                expected_registry: Registry {
                     scheme: "".to_string(),
                     insecure: false,
                     host: "quay.io".to_string(),
                     port: None,
                     namespace: "/ns1".to_string(),
                 },
-            ),
-        ];
-
-        for (input, expected) in tests {
-            let registry: Registry = Registry::try_from_str(input)
-                .unwrap_or_else(|_| panic!("could not parse {} to registry", input));
-            assert_eq!(registry, expected);
-            assert_eq!(input, registry.host_port_string());
+                expected_registry_host_port_string: "quay.io".to_string(),
+                expected_registry_host_port_namespaced_string: "quay.io/ns1".to_string(),
+            },
+        ]
+    }
+
+    #[test]
+    fn registry_try_parse_valid() {
+        let tests = setup_test_cases();
+        for test in tests {
+            let registry: Registry = Registry::try_from_str(&test.input)
+                .unwrap_or_else(|_| panic!("could not parse {} to registry", &test.input));
+            assert_eq!(test.expected_registry, registry);
+        }
+    }
+
+    #[test]
+    fn registry_host_port_string() {
+        let tests = setup_test_cases();
+        for test in tests {
+            let registry: Registry = Registry::try_from_str(&test.input)
+                .unwrap_or_else(|_| panic!("could not parse {} to registry", &test.input));
+            assert_eq!(
+                test.expected_registry_host_port_string,
+                registry.host_port_string()
+            );
```

Comment:
Not sure if we should rather test this against the expected_registry.

```suggestion
            assert_eq!(
                test.expected_registry_host_port_string,
                test.expected_registry.host_port_string()
            );
```

Reply:
Using `registry` as the tested value feels correct to me

Reply:
As the `TestCase` struct has the field `input`, I am going to continue using the `registry` as the tested value. It does not matter *that* much.